scripts/Diffraction/isis_powder/routines/sample_details.py

```diff
@@ -3,22 +3,31 @@
 from six import iteritems
 from isis_powder.routines import common
 import math
+from mantid import logger
+
+property_err_string = "The following sample property was not passed as an argument: {}"
 
 
 class SampleDetails(object):
     def __init__(self, **kwargs):
-        # By using kwargs we get a better error that, init takes 4 arguments
-        err_string = "A following sample property was not passed as an argument: "
-        height = common.dictionary_key_helper(dictionary=kwargs, key="height", exception_msg=err_string + "height")
-        radius = common.dictionary_key_helper(dictionary=kwargs, key="radius", exception_msg=err_string + "radius")
-        center = common.dictionary_key_helper(dictionary=kwargs, key="center", exception_msg=err_string + "center")
-
-        # Currently we only support cylinders
-        self.shape_type = "cylinder"
-        SampleDetails._validate_sample_details_constructor_inputs(height=height, radius=radius, center=center)
-        self.height = float(height)
-        self.radius = float(radius)
-        self.center = [float(i) for i in center]  # List of X, Y, Z position
+        self._shape_type = common.dictionary_key_helper(dictionary=kwargs, key="shape", throws=False)
+        if self._shape_type is None:
+            self._shape_type = "cylinder"
+            warning = "Failed to supply parameter \"shape\" - defaulting to \"cylinder\""
+            logger.warning(warning)
```

Comment:
Could we make this so that it also makes a `print` call here as well? Just so you get a message both in the scrit window and the log window?